### PR TITLE
[IMP] support translations

### PR DESCRIPTION
**Pull request by std-odoo**
Purpose
=======
This commit allow the users to translate their Gmail Addon.

The language of the addon will be the language the user set on his
preference, on his Odoo database.

The translations will be downloaded from Odoo and stored in the cache,
so translator can use the existing system to translate the addon.

Task 2502261
See /pull/8
See odoo/odoo/pull/69288
See odoo/enterprise/pull/17711

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
Could you rebase on mail-client-extensions master so that we can do a push-as-a-merge :) ?

**Comment by std-odoo** [comment]
Done :) still need to rebase the community branch

File: gmail/src/services/translation.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import { postJsonRpc } from "../utils/http";
+import { URLS } from "../const";
```
Comment (tde-banana-odoo) [inline gmail/src/services/translation.ts:2]: Linting: we could order alphabetically

File: gmail/src/services/translation.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import { State } from "../models/state";
+
+/**
+ * Object which fetch the translations on the Odoo database, put them in cache.
```
Comment (tde-banana-odoo) [inline gmail/src/services/translation.ts:6]: Object that fetches [ ... ] puts

File: gmail/src/const.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 const URLS: Record<string, string> = {
     LOG_EMAIL: "/mail_plugin/log_mail_content",
+    GET_TRANSLATIONS: "/mail_plugin/get_translations",
```
Comment (tde-banana-odoo) [inline gmail/src/const.ts:3]: Could be set before LOG_EMAIL (alphabetical ordering)

File: gmail/src/services/translation.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            if (this.translations) {
+                // Put in the cacher for 6 hours (maximum cache life time)
+                cache.put(cacheKey, JSON.stringify(this.translations), 21600);
+            }
+        }
+
+        this.translations = this.translations || {};
+    }
+
+    _t(text: string, parameters: any = undefined): string {
```
Comment (tde-banana-odoo) [inline gmail/src/services/translation.ts:40]: Could be great to add a docstring explaining format we support in translations.

**Review by tde-banana-odoo — APPROVED** [review]
LGTM, some linting remarks !

**Comment by std-odoo** [comment]
@tde-banana-odoo Changes done :)

**Pull request merged**